src/Csv/Decode.elm

```diff
@@ -629,38 +638,102 @@ errorToString error =
-                                ++ rowAndColumnString err
-                                ++ ": "
-                                ++ only
-
-                        many ->
-                            "There were some problems on "
-                                ++ rowAndColumnString err
-                                ++ ":\n\n"
-                                ++ String.join "\n" (List.map (\problem -> " - " ++ problem) many)
+                            case prevGroup of
+                                [] ->
+                                    List.reverse soFar
+
+                                head :: tail ->
+                                    List.reverse ({ startRow = List.reverse tail |> List.head |> Maybe.withDefault head |> .row, endRow = head.row, column = head.column, problem = head.problem } :: soFar)
+
+                        err :: rest ->
+                            case prevGroup of
+                                [] ->
+                                    dedupeHelp soFar (err :: prevGroup) rest
+
+                                head :: tail ->
+                                    if head.problem == err.problem && head.row + 1 == err.row && head.column == err.column then
+                                        dedupeHelp soFar (err :: prevGroup) rest
+
+                                    else
+                                        dedupeHelp ({ startRow = List.reverse tail |> List.head |> Maybe.withDefault head |> .row, endRow = head.row, column = head.column, problem = head.problem } :: soFar) [ err ] rest
+
+                dedupeErrs : List { row : Int, column : Column, problems : List Problem } -> List { startRow : Int, endRow : Int, column : Column, problem : Problem }
+                dedupeErrs =
+                    List.concatMap (\err -> List.map (\problem -> { row = err.row, column = err.column, problem = problem }) err.problems)
+                        >> List.sortBy
+                            (\{ row, problem } ->
+                                case problem of
+                                    ColumnNotFound i ->
+                                        ( 1, "", row )
+
+                                    FieldNotFound name ->
+                                        ( 2, name, row )
+
+                                    FieldNotProvided name ->
+                                        ( 3, name, row )
+
+                                    ExpectedOneColumn howMany ->
+                                        ( 4, String.fromInt howMany, row )
+
+                                    ExpectedInt notInt ->
+                                        ( 5, notInt, row )
+
+                                    ExpectedFloat notFloat ->
+                                        ( 6, notFloat, row )
+
+                                    Failure custom ->
+                                        ( 7, custom, row )
+                            )
+                        >> dedupeHelp [] []
```

Comment:
could we sort by start row instead of error type? If we're improving the problem-solving experience with a CSV, I think it'd be nicer for someone to be able to work the error list top-to-bottom instead of by type.